kasenvr/project-athena
# Add support for the Opus audio plugin.

This PR creates and enabled support for the Opus audio codec in the interface and audio mixer. Some details include:
- Current bitrate is 128000 bits per second. Further testing for audio quality will probably be necessary.
- This codec shifts resources; it uses lower bandwidth but increases CPU use.
- The parameters that determine codec quality are currently hard coded. Further work will be needed to allow domain operators to specify Opus profiles.

To test:
- Compile! Or use a PR build if available.
- Run a domain-server and the associated assignment-clients.
- On the domain-server web console, go to Audio Environment, Advanced Settings, and set the Audio Codec Preference Order so that `opus` appears as the only available codec.
- Start the interface, and enter the domain we just created.
- Open the interface's stats (press the / key). 
- Further expand statistic details by left clicking in any stat pane.
- At this point, you should see the opus codec as the one currently in use.

<img width="960" alt="Opus_PR" src="https://user-images.githubusercontent.com/464056/72290315-8496ad80-3612-11ea-9ce1-2f249ec50dc6.png">
